bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+        }
+        logger.debug("Finding properties for device {} using salusClient", dsn);
+        var response = api.findDeviceProperties(dsn);
+        if (response.failed()) {
+            logger.error("Cannot find properties for device {} using salusClient\n{}", dsn, response.error());
+            return null;
+        }
+        return response.body();
+    }
+
+    @Override
+    public boolean setValueForProperty(String dsn, String propertyName, Object value) {
+        var api = salusApi;
+        if (api == null) {
+            logger.error("Cannot set value for property {} on device {} because salusClient is null", propertyName,
+                    dsn);
+            return false;
+        }
+        logger.debug("Setting property {} on device {} to value {} using salusClient", propertyName, dsn, value);
+        var response = api.setValueForProperty(dsn, propertyName, value);
+        if (response.failed()) {
+            logger.error("Cannot set property {} on device {} to value {} using salusClient\n{}", propertyName, dsn,
+                    value, response.error());
+            devicePropertiesCache.invalidate(dsn);
+            return false;
+        }
+        var setValue = response.body();
+        if (setValue != null
+                && (!(setValue instanceof Boolean) && !(setValue instanceof String) && !(setValue instanceof Number))) {
+            logger.warn(
+                    "Cannot set value {} ({}) for property {} on device {} because it is not a Boolean, String, Long or Integer",
+                    setValue, setValue.getClass().getSimpleName(), propertyName, dsn);
+            return false;
+        }
+        var property = devicePropertiesCache.get(dsn).stream().filter(prop -> prop.getName().equals(propertyName))
+                .findFirst();
+        if (property.isEmpty()) {
+            var simpleName = setValue != null ? setValue.getClass().getSimpleName() : "<null>";
+            logger.warn(
+                    "Cannot set value {} ({}) for property {} on device {} because it is not found in the cache. Invalidating cache",
+                    setValue, simpleName, propertyName, dsn);
+            devicePropertiesCache.invalidate(dsn);
+            return false;
+        }
+        var prop = property.get();
+        if (setValue == null) {
+            prop.setValue(null);
+            return true;
+        }
+        if (setValue instanceof Boolean b && prop instanceof DeviceProperty.BooleanDeviceProperty boolProp) {
+            boolProp.setValue(b);
+            return true;
+        }
+        if (setValue instanceof String s && prop instanceof DeviceProperty.StringDeviceProperty stringProp) {
+            stringProp.setValue(s);
+            return true;
+        }
+        if (setValue instanceof Number l && prop instanceof DeviceProperty.LongDeviceProperty longProp) {
+            longProp.setValue(l.longValue());
+            return true;
+        }
+
+        logger.warn(
```

Comment:
```suggestion
        logger.debug(
```

Reply:
Are you sure it should be debug?

Reply:
This is similar case like this https://github.com/openhab/openhab-addons/pull/16065/files#r1460476871

Reply:
should I close this (same use case as https://github.com/openhab/openhab-addons/pull/16065#discussion_r1460575819 )?

Reply:
ping